src/QueryAccess/Condition.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\entity\QueryAccess;
+
+/**
+ * Represents a single access condition.
+ */
+final class Condition {
+
+  /**
+   * The supported operators.
+   *
+   * @var array
+   */
+  protected static $supportedOperators = [
+    '=', '<>', '<', '<=', '>', '>=', 'BETWEEN', 'NOT BETWEEN',
+    'IN', 'NOT IN', 'IS NULL', 'IS NOT NULL',
+  ];
+
+  /**
+   * The field.
+   *
+   * @var string
+   */
+  protected $field;
+
+  /**
+   * The value.
+   *
+   * @var mixed
+   */
+  protected $value;
+
+  /**
+   * The operator.
+   *
+   * @var string
+   */
+  protected $operator;
+
+  /**
+   * Constructs a new Condition object.
+   *
+   * @param string $field
+   *   The field, with an optional column name. E.g: 'uid', 'address.locality'.
+   * @param mixed $value
+   *   The value.
+   * @param string $operator
+   *   The operator.
+   *   Possible values: =, <>, <, <=, >, >=, BETWEEN, NOT BETWEEN,
+   *                   IN, NOT IN, IS NULL, IS NOT NULL.
+   */
+  public function __construct($field, $value, $operator = NULL) {
+    // Provide a default based on the data type of the value.
```

Comment:
Because that's what core does in its own conditions (SQL / EntityQuery), and I wanted the condition API to be similar. I could make the operator required and move the default setting to ConditionGroup::addCondition() but I'm not sure that helps. And if there's no default on any level, then the operator must be specified even when it's '=', which makes it inconsistent with core.

Reply:
Core also has far too many "magic defaults". But I defer to you to decide on this one — I don't feel very strongly about it. I'm just pointing out the addition of new magic. But if that magic matches what core already does, and it does so for good reason, then 👍

Reply:
Hm, looking at this again, core has `\Drupal\Core\Database\Query\ConditionInterface`, which is _very_ different from the `Condition` and `ConditionGroup` classes being added here. Neither of those classes even implements core's interface. I understand why they don't. But then I don't understand why we want it to be as similar as possible if it's already so dissimilar?

Reply:
Decided to punt on this for now, but if it gets raised again (core review or here) I have nothing against changing it.